content/200-concepts/100-components/03-prisma-migrate/010-mental-model.mdx

```diff
@@ -0,0 +1,137 @@
+
+![Prisma Migrate "state management"](https://user-images.githubusercontent.com/33921841/189885586-0e0d7852-8ac7-4842-a6cb-bd158c20d28e.png)
+
+## Evolve your database schema with Prisma Migrate
+
+### Assumptions when working with Prisma Migrate
+
+- Ideally, you should use different databases for different environments. For example, you might have a separate database for development, preview, and production environments.
+- The databases you use in development environments are disposable — you can easily create, use, and delete on demand.
+- The database configuration used in different environments should be similar to ensure it is reproducible.
+- The Prisma schema serves as the source of truth — describing the shape of your [database schema](https://www.prisma.io/dataguide/intro/database-glossary#schema).
+
+### Concepts
+
+- [Schema drift](/guides/database/developing-with-prisma-migrate/troubleshooting-development#schema-drift): occurs when your database schema is out of sync with your migrations history and migrations table.
+
+### Development environments
+
+#### Prototype your schema
+
+Prisma Migrate's [`db push` command](/reference/api-reference/command-reference#db-push) allows you to sync your Prisma schema and database schema without persisting a migration (`/prisma/migrations`). The `db push` command uses the **Prisma schema** and the **database schema** to track the database state.
+
+![prisma db push development flow](https://user-images.githubusercontent.com/33921841/189866483-cec55d83-d1fe-4446-9c28-0a60329f975d.png)
+
+The `db push` command is useful for quickly prototyping and iterating on your database schema and prioritizing a desired end state for your database schema.
+
+If the `db push` command detects a schema drift, it will prompt you to reset your database. For example, this will happen when you add a required field to a table with existing content without providing a default value.
+
+#### Track your migration history with `migrate dev`
+
+Prisma Migrate's [`migrate dev` command](/reference/api-reference/command-reference#migrate-dev) allows you to track the changes you make to your database. The `migrate dev` command automatically generates SQL migration files (saved in `/prisma/migrations`) and applies them to the database. When a migration is applied to the database, the migrations table (`_prisma_migrations`) in your database is also updated.
+
+![Prisma Migrate dev flow](https://user-images.githubusercontent.com/33921841/189875756-5df6ba94-c7a9-4e9c-a8f7-ce4d5d09b204.png)
+
+The `migrate dev` command tracks the database' state using the:
+
+- Prisma schema
+- Migrations history
+- Migrations table
+- Database schema
+
+You can customise migrations before you apply them to the database using the `--create-only` flag. For example, you might want to edit a migration if you want to rename columns without incurring any data loss or load database extensions (in PostgreSQL) and database views.
```

Comment:
Is the idea of these examples that they are things that Prisma does not currently support? Probably worth spelling that out if so.

```suggestion
You can customise migrations before you apply them to the database using the `--create-only` flag. For example, you might want to edit a migration if you want to rename columns without incurring any data loss or load database extensions (in PostgreSQL) and database views.
```